### PR TITLE
Do not wait for re-balance to complete if not needed

### PR DESCRIPTION
**Pull request by ebenahar**
In test cases that cause PGs rebalance, instead of waiting for rebalance to complete, this change suggests moving the next test case to be executed last and move on to a test case that does not require PGs to be active+clean

Signed-off-by: Elad Ben Aharon <ebenahar@redhat.com>

### CODE REVIEW

File: tests/manage/z_cluster/nodes/test_nodes_maintenance.py [hunk 1]
```diff
@@ -80,6 +80,8 @@ class TestNodesMaintenance(ManageTest):
 
     """
 
+    REBALANCE_COMPLETED_NEEDED = True
```
Comment (yitzhak12) [inline tests/manage/z_cluster/nodes/test_nodes_maintenance.py:83]: Why did you put it in this file?

File: tests/conftest.py [hunk 2]
```diff
@@ -204,6 +205,30 @@ def pytest_collection_modifyitems(session, items):
...
+    completed, in case PG re-balance completion is needed for it
+    """
+    yield
+    idx = item.session.items.index(item)
+    remaining = item.session.items[idx + 1 :]
+    if nextitem.cls.REBALANCE_COMPLETED_NEEDED:
+        cluster = CephCluster()
+        rebalance_status = cluster.get_rebalance_status()
+        if not rebalance_status:
+            if config.RUN["first_pushed_to_last_item"] is not nextitem:
```
Comment (yitzhak12) [inline tests/conftest.py:221]: Can you explain more about the meaning of this condition?

**Comment by petr-balogh** [comment]
I think that this also has to be well documented - so please add it to some of the documentation file.

**Review by mbukatov — COMMENTED** [review]
The ocs-ci test are neither classified nor isolated so that we know which test cases requires particular level of cluster access. Letting this in current state would only increase effort required for CI analysis, and created unnecessary test failures.

I'm sure I discussed this somewhere already, but I'm adding it here since it was brought up today.

**Comment by ebenahar** [comment]
Re-opening for re-examination

**Comment by petr-balogh** [comment]
@ebenahar can this be closed or still relevant?

**Pull request closed without merging**